### PR TITLE
made changes to username functionality

### PR DESCRIPTION
**Pull request by nuxeric**
# Changes
I made an ExtendedSocket interface to use to type all sockets in the backend
I generate the names and set their state in the frontend

# Tickets
#82

### CODE REVIEW

File: backend/src/core/server.ts [hunk 1]
```diff
@@ -50,26 +40,21 @@ export default class Server {
 
   private setupSockets(): void {
     const io = socketIo(this.httpServer);
-    io.on(Event.CONNECT, socket => {
+    io.on(Event.CONNECT, (socket: ExtendedSocket) => {
+      socket.username = "";
```
Comment (cowmanjoe) [inline backend/src/core/server.ts]: Why is this necessary? If it is to satisfy the Typescript type in some way maybe we could change it to be optional (i.e. `username?: string` in the `ExtendedSocket` interface

File: frontend/src/components/Room.tsx [hunk 2]
```diff
@@ -163,11 +171,27 @@ class Room extends React.Component<Props, State> {
...
+      this.setState(
+        {
+          userName: randomName
+        },
+        () => {
+          this.socket.emit(Event.CREATE_USERNAME, this.state.userName);
+        }
+      );
+    } else {
+      this.socket.emit(Event.CREATE_USERNAME, this.state.userName);
```
Comment (cowmanjoe) [inline frontend/src/components/Room.tsx]: I think `username` is one word so there should be no capital `N` in the middle

File: backend/src/core/server.ts [hunk 3]
```diff
@@ -50,26 +40,21 @@ export default class Server {
...
-      socket.on(Event.CREATE_USERNAME, username => {
-        const extSocket = socket as ExtendedSocket;
-        if (username === "") {
-          const randomName: string = uniqueNamesGenerator(customNameConfig);
-          extSocket.username = `Anonymous ${randomName}`;
-        } else {
-          extSocket.username = username;
-        }
-        logger.info(`socket username set to ${extSocket.username}`);
+      socket.on(Event.CREATE_USERNAME, (username: string) => {
```
Comment (cowmanjoe) [inline backend/src/core/server.ts]: I think this should probably be in `handler.ts`

File: backend/src/core/server.ts (hunk 1 above)
Comment (nuxeric) [reply backend/src/core/server.ts]: You're right I made the username optional and it satisfied the Typescript typing thank you

File: frontend/src/components/Room.tsx (hunk 2 above)
Comment (nuxeric) [reply frontend/src/components/Room.tsx]: Done! :D

File: backend/src/core/server.ts (hunk 3 above)
Comment (nuxeric) [reply backend/src/core/server.ts]: Moved! :D thank you
